tap_core/src/checks/mod.rs

```diff
@@ -0,0 +1,250 @@
+// Copyright 2023-, Semiotic AI, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::tap_receipt::{Checking, ReceiptError, ReceiptResult, ReceiptWithState};
+use serde::{Deserialize, Serialize};
+use std::sync::{Arc, RwLock};
+
+pub type ReceiptCheck = Arc<dyn Check>;
+
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub enum CheckingChecks {
```

Comment:
`CheckingChecks` is a pretty weird name. Would something more direct such as `CheckWithState` make sense?

Reply:
I'll remove in the next PR, so it doesn't make sense to rename now.